src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/RealmCommand.scala

```diff
@@ -0,0 +1,32 @@
+package ch.epfl.bluebrain.nexus.iam.realms
+import cats.effect.Async
+import ch.epfl.bluebrain.nexus.iam.realms.RealmEvent._
+import ch.epfl.bluebrain.nexus.iam.realms.RealmRejection._
+import ch.epfl.bluebrain.nexus.iam.realms.RealmState._
+
+sealed trait RealmCommand extends Product with Serializable
+
+object RealmCommand {
+
+  final case class CreateRealm(realm: Realm)           extends RealmCommand
+  final case class UpdateRealm(realm: Realm, rev: Int) extends RealmCommand
+  final case class DeprecateRealm(rev: Int)            extends RealmCommand
+
+  def evaluate[F[_]](state: RealmState, command: RealmCommand)(
+      implicit F: Async[F]): F[Either[RealmRejection, RealmEvent]] = (state, command) match {
+    case (Initial, CreateRealm(realm))      => F.pure(Right(RealmCreated(realm)))
```

Comment:
The realm is not always valid by default. Here you will have to attempt to [fetch the PublicKey](https://github.com/BlueBrain/nexus-iam/blob/b2ce115abc1386311471a48a47ae10811bb0cb11/modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/auth/CredentialsStoreActor.scala#L61) and pass it to the state or reject when not it does not pass.